### ISSUE TITLE
Bug: CoreBlocks doesn't work with TypeScript

### ISSUE THREAD

**Issue by traed** [wpengine/faustjs#1729]
## Environment
- `@faustwp/core` version: 2.1.0
- `@faustwp/cli` version: 2.0.0
- `@faustwp/blocks` version: 2.0.0

## Description
Core blocks imported from @faustwp/blocks doesn't implement the WordPressBlock interface and can therefore not be used with the WordPressBlocksProvider component in a TypeScript app since WordPressBlocksProviderConfig requires a list of WorkPressBlock's.

If I change CoreParagraphFragmentProps and make `attributes` optional it works.

Attempting to use the CoreParagraph block, for example, gives the following error in my editor:

```
Type 'typeof CoreParagraph' is not assignable to type 'WordPressBlock<{ [key: string]: any; }>'.
  Type 'typeof CoreParagraph' is not assignable to type 'FC<{ [key: string]: any; }>'.
    Types of parameters 'props' and 'props' are incompatible.
      Type '{ [key: string]: any; }' is not assignable to type 'CoreParagraphFragmentProps'.
        Property 'attributes' is missing in type '{ [key: string]: any; }' but required in type '{ attributes: { cssClassName?: string | undefined; backgroundColor?: string | undefined; content?: string | undefined; style?: string | undefined; textColor?: string | undefined; fontSize?: string | undefined; ... 4 more ...; align?: string | undefined; }; }'.ts(2322)
CoreParagraph.d.ts(4, 5): 'attributes' is declared here.
WordPressBlocksProvider.d.ts(30, 9): The expected type comes from this index signature.
```
My _app.tsx:
```
import React from "react";
import { useRouter } from "next/router";
import { FaustProvider } from "@faustwp/core";
import { AppProps } from "next/app";
import { CoreBlocks, WordPressBlocksProvider } from "@faustwp/blocks";

export default function MyApp({ Component, pageProps }: AppProps) {
  const router = useRouter();

  return (
    <FaustProvider pageProps={pageProps}>
      <WordPressBlocksProvider
        config={{ blocks: { CoreParagraph: CoreBlocks.CoreParagraph } }}
      >
        <Component {...pageProps} key={router.asPath} />
      </WordPressBlocksProvider>
    </FaustProvider>
  );
}

```

**Comment by theodesp** [comment]
Hey @traed sorry for not responding earlier.

I'm not seeing this type error when I use the [faust-scafold-ts](https://github.com/wpengine/faust-scaffold-ts) project.

AFIK WordPressBlocksProviderConfig does accept an object and not a list of 

```
export type WordPressBlocksProviderConfig = {
    blocks: {
        [key: string]: WordPressBlock;
    };
    theme?: BlocksTheme;
};
```

Where:

```
export type WordPressBlock<P = {
    [key: string]: any;
}> = FC<P> & Partial<Pick<WordPressBlockBase, 'config' | 'displayName' | 'name' | 'fragments'>>;
```
So it should allow for an object with `attributes` field.

Also I noticed that we already have a type that makes the attributes optional:

https://github.com/wpengine/faustjs/blob/canary/packages/blocks/src/components/WordPressBlocksViewer.tsx#L38-L40
```
export interface BlockWithAttributes extends ContentBlock {
    attributes?: Record<string, unknown> | null;
}
```
Do you think that something needs to be updated?

**Comment by traed** [comment]
By "list" I meant a collection of blocks, didn't specify if it was an array or object, but that's not important.

The problem is that in the CoreParagraphFragmentProps interface (and the other core blocks) the attributes prop is required. There is no **required** prop named attributes in BlockWithAttributes or WordPressBlock, hence the error.

The updates I think are needed is in the PR I linked here, namely that the attributes prop in the core block interfaces needs to be optional.

**Comment by theodesp** [comment]
Hey @traed how exactly are you using the block since I cannot reproduce locally?

**Comment by traed** [comment]
I posted my `_app.tsx` file in my original post. That is the only place it is referenced currently.

Also, I took a closer look at the scaffold project you linked and that has `"strict": false` in `tsconfig.json` but my local copy had `"strict": true`. Setting strict mode to false suppresses the error so I guess that is why you don't see it.

**Comment by theodesp** [comment]
Hey @traed thank you. We will take a look at your suggestion.

**Comment by theodesp** [comment]
@traed thanks for the suggestion. We can merge your PR once the 

https://github.com/wpengine/faustjs#contributor-license-agreement process is completed

**Comment by blakewilson** [comment]
Hi @traed, thank you for your PR. Your changes have been released in @faustwp/blocks@3.0.0. You can view the changelog here:

https://github.com/wpengine/faustjs/releases/tag/%40faustwp%2Fblocks%403.0.0
